code/datums/martial/karate.dm

```diff
@@ -0,0 +1,123 @@
+#define CALF_KICK_COMBO "HGD" //Calf Kick - paralyse one leg
+#define FLOOR_KICK_COMBO "HGH" //Floor Stomp - brute and stamina damage if target isn't standing
+#define JUMPING_KNEE_COMBO "HDH" //Jumping Knee - knockdown and stamina damage
+#define KARATE_CHOP_COMBO "GHD" //Karate Chop - short confusion and blurred eyes
+
+/datum/martial_art/karate
+	name = "Karate"
+	id = MARTIALART_KARATE
+	allow_temp_override = FALSE
+	help_verb = /mob/living/carbon/human/proc/karate_help
+	display_combos = TRUE
+
+/datum/martial_art/karate/proc/check_streak(mob/living/carbon/human/A, mob/living/carbon/human/D)
```

Comment:
just as a suggestion but you can remove the single letter vars to better explain who A and D is throughout the whole file

Reply:
Noted. I'll think about it

Reply:
I'd agree if literally all martial arts weren't coded this way.